src/ViewLayer/View.h

```diff
@@ -0,0 +1,41 @@
+#ifndef VIEW_H
+#define VIEW_H
+
+#include <QObject>
+class TelemetryReporting;
+class QSerialPort;
+class Window;
+class QPushButton;
+class QComboBox;
+class QLineEdit;
+class QLabel;
+
+class View : public QObject
+{
+    Q_OBJECT
+public:
+    View(QSerialPort& serialPort, TelemetryReporting& telemetryReporting);
+
+private:
+    void connectSlots();
+
+private slots:
+    void attemptConnection();
+    void differentModeSelected();
+    void sendKeyDriverControl();
+    void sendDriverControlDetails();
+    void sendFaults();
+    void sendBatteryData();
+    void sendCmuData();
+    void sendMpptData();
+    void sendAll();
```

Comment:
So later, these should be signals which get sent to the reporting layer, which activate the slot to do the logical stuff.